pre_commit/commands/run.py

```diff
@@ -37,7 +43,15 @@ def get_changed_files(new, old):
     )[1].splitlines()
 
 
-def get_filenames(args, include_expr, exclude_expr):
+def filter_filenames_by_types(filenames, types):
+    types = frozenset(types)
+    return tuple(
+        filename for filename in filenames
+        if tags_from_path(filename) >= types
```

Comment:
Yeah, I can't really come up with a real use-case for either. It's still possible to get the OR behavior by specifying the hook twice if you really need it. I'm fine with either way.

fwiw all of the standard hooks were easy to write with just one type: https://github.com/chriskuehl/pre-commit-hooks/blob/new-typers/hooks.yaml